src/pyqasm/elements.py

```diff
@@ -98,3 +98,20 @@ class Variable:  # pylint: disable=too-many-instance-attributes
     is_constant: bool = False
     is_register: bool = False
     readonly: bool = False
+
+
+class BasisSet(Enum):
+    """
+    Enum for the different basis sets in Qasm.
+    """
+
+    DEFAULT = 0
+    ROTATIONAL_CX = 1
+    CLIFFORD_T = 2
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}.{self.name}: {self.value}>"
```

Comment:
This can be deleted. `Enum` classes already come with a well-implemented `repr` method specifying the class name, member name, and member value, so we don't need a custom one.

For example:

```python
from enum import Enum

class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3

print(repr(Color.RED)) # <Color.RED: 1>
```